main.go

```diff
@@ -16,6 +16,25 @@ import (
 	_ "github.com/heroku/x/hmetrics/onload"
 )
 
+var (
+	downloadConcurrency  int
+	maxPayloadSizeInByte int64
+)
+
+func init() {
+	downloadConcurrency, _ = strconv.Atoi(os.Getenv("DOWNLOAD_CONCURRENCY"))
```

Comment:
I recommend splitting this step into two.
Can we continue with default value if `os.Getenv("DOWNLOAD_CONCURRENCY")` is `""`, i.e., unset,
and
abort if we find unexpected value in `DOWNLOAD_CONCURRENCY` (i.e., `err != nil` for `strconv.Atoi()`)?

Reply:
Could you explain the benefits?

Reply:
Just a nice to have.

1. If the value is not set, we'll be going with the default value.
2. If the value set, but is invalid, shouldn't we know about the issue?

Maybe we need not abort it, but maybe we could just log it?

Reply:
@arjunmayilvaganan, In either case, we need to check and ser the value. The default value is there just as a backup.